File: NetFlexAPI/GiraffeAPI/Models/Role.fs
module GiraffeAPI.Models.Role

open System
open System.ComponentModel.DataAnnotations
open Microsoft.EntityFrameworkCore.Metadata.Internal

[<CLIMutable>]  
type Role =
    {
        RoleId : Guid
        RoleName : string
    }

    member this.HasErrors =
        if this.RoleName = null || this.RoleName = "" then Some "Name of role is required"
        else None
        

File: NetFlexAPI/TestProject1/Role.cs
using System;

namespace TestProject;

public class Role
{
    public Guid RoleId { get; set; }
    public string RoleName { get; set; }

    public Role(Guid roleId, string roleName)
    {
        RoleId = roleId;
        RoleName = roleName;
    }
}